packages/tre/src/plugins/r2/router.ts

```diff
@@ -1,50 +1,188 @@
-import { Response } from "undici";
+import { TextDecoder } from "util";
+import { Request, Response } from "undici";
+import { GET, PUT, RouteHandler, Router, decodePersist } from "../shared";
+import { InternalError, R2Error } from "./errors";
 import {
-  DELETE,
-  GET,
-  PUT,
-  RouteHandler,
-  Router,
-  decodePersist,
-} from "../shared";
-import { R2Gateway } from "./gateway";
+  R2Gateway,
+  R2GetOptions,
+  R2ListOptions,
+  R2PutOptions,
+} from "./gateway";
+import { R2HTTPMetadata, R2Object } from "./r2Object";
 
+export enum CfHeader {
+  Error = "cf-r2-error",
+  Request = "cf-r2-request",
+  MetadataSize = "cf-r2-metadata-size",
+}
 export interface R2Params {
   bucket: string;
-  key: string;
 }
-export class R2Router extends Router<R2Gateway> {
-  @GET("/:bucket/:key")
-  get: RouteHandler<R2Params> = async (req, params) => {
-    // console.log(await req.json());
+const decoder = new TextDecoder();
 
-    const persist = decodePersist(req.headers);
-    const gateway = this.gatewayFactory.get(params.bucket, persist);
-    await gateway.get(params.key);
-    return new Response();
+const decodeMetadata = async (req: Request) => {
+  const bytes = await req.arrayBuffer();
+
+  const metadataSize = Number(req.headers.get(CfHeader.MetadataSize));
```

Comment:
I've added this as a safety check. The reason I didn't add it initially is because the runtime should unconditionally add this header with valid data (and if not, it's probably a bug in the runtime, rather than a code path that a user could trigger)